### PR TITLE
Struct-ifying schema things

### PR DESCRIPTION
**Pull request by yaleman**
I can't remember how I started doing this but I'm sure I was avoiding something more important...

* Rebuilt most of the schema things as structs instead of JSON string blobs
* Added build-time validation that we're not re-using UUIDs

Checklist

- [x] This pr contains no AI generated code
- [x] cargo fmt has been run
- [x] cargo clippy has been run
- [x] cargo test has been run and passes
- [ ] book chapter included (if relevant)
- [ ] design document included (if relevant)

### CODE REVIEW

File: server/lib/build.rs [hunk 1]
```diff
@@ -12,4 +14,27 @@ fn main() {
     }
 
     profiles::apply_profile();
+
+    // check we don't have duplicate UUIDs
```
Comment (Firstyear) [inline server/lib/build.rs:18]: This is clever, I really like this!

**Review by Firstyear — APPROVED** [review]
This is fantastic! Now how I would have approached it but still an excellent improvement and I really like what you've done here :)

**Comment by yaleman** [comment]
> This is fantastic! Now how I would have approached it but still an excellent improvement and I really like what you've done here :)

I was pondering last night making `Schema` like this:

```rust
enum SchemaType {
    SchemaAttribute,
    SchemaClass,
}


struct Schema {
    uuid: Uuid,
    name: &'static str,
    description: &'static str
    schema_type: SchemaType,
    attrs: SchemaAttrs,
}
```

Which would allow us to automagically do the `uuid/name/description/class` fields and set things up... how would you have done it? I'm happy to rewrite it again, this was just a mad-lad first rewrite.

**Comment by Seba-T** [comment]
Wow this looks marvelous! it feels like living 10 yrs into the future :p

**Comment by Firstyear** [comment]
> Which would allow us to automagically do the `uuid/name/description/class` fields and set things up... how would you have done it? I'm happy to rewrite it again, this was just a mad-lad first rewrite.

I probably would have just used the entry_init! macro, but I think this is a bit nicer/structured. :)

**Comment by Firstyear** [comment]
> I was pondering last night making `Schema` like this:

This is a good idea, then you can have the enum for attr/class have the specific bits for those types.

**Comment by yaleman** [comment]
> > I was pondering last night making `Schema` like this:
> 
> This is a good idea, then you can have the enum for attr/class have the specific bits for those types.

Y'know what's even smarter... USING THE STRUCTS WE ALREADY HAD! (done)

**Pull request merged**